package/MDAnalysis/coordinates/TRC.py

```diff
@@ -0,0 +1,399 @@
+                    elif "END" in coords_str:
+                        break
+                    else:
+                        tmp_buf.append(coords_str.split())
+
+                if (np.array(tmp_buf).shape[0] == self.n_atoms):
+                    frameDat["positions"] = np.asarray(tmp_buf,
+                                                       dtype=np.float64)
+                else:
+                    raise ValueError("The trajectory contains \
+                                     the wrong number of atoms!")
+
+            elif ("GENBOX" == line.strip()):
+                ntb_setting = int(f.readline())
+                if (ntb_setting == 0):
+                    frameDat["dimensions"] = None
+                    self.periodic = False
+
+                elif (ntb_setting in [-1, 1]):
+                    tmp_a, tmp_b, tmp_c = f.readline().split()
+                    tmp_alpha, tmp_beta, tmp_gamma = f.readline().split()
+                    frameDat["dimensions"] = [float(tmp_a),
+                                              float(tmp_b),
+                                              float(tmp_c),
+                                              float(tmp_alpha),
+                                              float(tmp_beta),
+                                              float(tmp_gamma)]
+                    self.periodic = True
+
+                    line3 = f.readline().split()
+                    line4 = f.readline().split()
+                    for v in (line3 + line4):
+                        if (float(v) != 0.0):
+                            raise NotImplementedError("This reader \
+                                                      supports neither \
+                                                      triclinic and/or \
+                                                      (yawed,pitched, \
+                                                      rolled) boxes!")
+                else:
+                    raise NotImplementedError("This reader does only support\
+                                               vacuum and rectangular boxes!")
+                break
+
+            elif any(non_supp_bn in line for
+                     non_supp_bn in self.NOT_SUPPORTED_BLOCKNAMES):
+                for non_supp_bn in self.NOT_SUPPORTED_BLOCKNAMES:
+                    if (non_supp_bn == line.strip()):
+                        warnings.warn(non_supp_bn + " block is not supported!",
+                                      UserWarning)
+
+        return frameDat
+
+    def _read_frame(self, i):
+        """read frame i"""
+        self._frame = i - 1
+
+        # Move position in file just (-2 byte) before the start of the block
+        self.trcfile.seek(self.traj_properties["l_blockstart_offset"][i]-2, 0)
+
+        return self._read_next_timestep()
+
+    def _read_next_timestep(self):
+
+        self._frame += 1
+        if (self._frame >= self.n_frames):
+            raise EOFError('Trying to go over trajectory limit')
```

Comment:
Technically, EOFError is correct but we chose to use IOError. Please change to be consistent.

Reply:
This case should also be tested.